### PR TITLE
update unifi and influxdb; fix unifi db creation Fixes issue #151

### LINKED ISSUE #151: Unifi database not created in InfluxDB
# Issue Template

## Description
When creating the Unifi jail w/ integrated Controller, the unifi database is not being created in Influx.

### Steps to Reproduce

1. Create Influx jail and Unifi Jail 
2.  Observe no unifi database being created
3.  

Noticed this when testing Grafana. I'll take a look when I get to the Unifi fixes.

### PR DESCRIPTION
**Pull request by colemamd**
# Pull Request Template

### Purpose
Fixes #151 

### Notes:
Tested until rebase to dev, now getting the same Issue #175 , marked as Draft until can retest when that's fixed.

### Please make sure you have followed the self checks below before submitting a PR:

- [X] Code is sufficiently commented
- [X] Code is indented with tabs and not spaces
- [] The PR does not bring up any new errors
- [] The PR has been tested
- [X] Any new files are named using lowercase (to avoid issues on case sensitive file systems)

### CODE REVIEW

File: blueprints/influxdb/install.sh [hunk 1]
```diff
@@ -15,7 +15,6 @@ iocage exec "${1}" sysrc influxd_enable="YES"
 
 # Start influxdb and wait for it to startup
 iocage exec "${1}" service influxd start
-sleep 15
 
 exitblueprint "$1" "You may connect InfluxDB plugins to the InfluxDB jail at http://${ip4_addr%/*}:8086."
```
Comment (Ornias1993) [inline blueprints/influxdb/install.sh:19]:
```suggestion
exitblueprint "$1" "You may connect InfluxDB plugins to the InfluxDB jail at http://${jail_ip}:8086."
```

File: blueprints/unifi/config.yml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 blueprint:
   unifi:
     pkgs: jq unifi5
-    traefik_service_port: 8443
-    vars: link_influxdb unifi_poller influxdb_database influxdb_user influxdb_password poller_user poller_password
+    vars: link_influxdb unifi_poller database database_user database_password poller_user poller_password
```
Comment (Ornias1993) [inline blueprints/unifi/config.yml:4]: no.

File: blueprints/unifi/install.sh [hunk 3]
```diff
@@ -5,14 +5,10 @@
 initblueprint "$1"
 
 # Initialize variables
-influxdb_database="${influxdb_database:-$1}"
-influxdb_user="${influxdb_user:-$influxdb_database}"
+database="${database:-$1}"
+database_user="${database_user:-$database}"
```
Comment (Ornias1993) [inline blueprints/unifi/install.sh:9]: no

File: blueprints/unifi/install.sh [hunk 4]
```diff
@@ -23,11 +19,11 @@ cp "${includes_dir}"/rc/unifi.rc /mnt/"${global_dataset_iocage}"/jails/"${1}"/ro
...
 iocage exec "${1}" service unifi start
 
-if [[ ! "${poller}" ]]; then
+if [ "${poller}" == "false" ]; then
   echo "Unifi Poller not selected, skipping Unifi Poller installation."
 else
-	if [ -z "${influxdb_password}" ]; then
-	echo "influxdb_password can't be empty"
+	if [ -z "${database_password}" ]; then
+	echo "database_password can't be empty"
```
Comment (Ornias1993) [inline blueprints/unifi/install.sh:26]: no

File: blueprints/unifi/install.sh [hunk 5]
```diff
@@ -23,11 +19,11 @@ cp "${includes_dir}"/rc/unifi.rc /mnt/"${global_dataset_iocage}"/jails/"${1}"/ro
 iocage exec "${1}" sysrc unifi_enable=YES
 iocage exec "${1}" service unifi start
 
-if [[ ! "${poller}" ]]; then
+if [ "${poller}" == "false" ]; then
```
Comment (Ornias1993) [inline blueprints/unifi/install.sh:22]:
```suggestion
if [ -z "${poller}" ]; then
```

File: blueprints/unifi/install.sh [hunk 6]
```diff
@@ -5,14 +5,10 @@
...
 # Initialize variables
-influxdb_database="${influxdb_database:-$1}"
-influxdb_user="${influxdb_user:-$influxdb_database}"
+database="${database:-$1}"
+database_user="${database_user:-$database}"
 poller_user="${poller_user:-$1}"
 
-#TODO LINK
-DB_IP="jail_${link_influxdb}_ip4_addr"
-DB_IP="${!DB_IP%/*}"
```
Comment (Ornias1993) [inline blueprints/unifi/install.sh:14]: Ahh shoot, yes this has been changed for some time now, sorry...

File: blueprints/unifi/install.sh [hunk 7]
```diff
@@ -44,18 +40,18 @@ else
...
+      echo "${link_influxdb} jail exists, but database ${database} does not. Creating database ${database}."
+      if [[ -z "${database_password}" ]]; then
+        echo "Database password not provided. Cannot create database without credentials. Exiting..."
         exit 1
       else
         # shellcheck disable=SC2027,2086
-        iocage exec "${link_influxdb}" "curl -XPOST -u ${influxdb_user}:${influxdb_password} http://"${DB_IP}":8086/query --data-urlencode 'q=CREATE DATABASE ${influxdb_database}'"
-        echo "Database ${influxdb_database} created with username ${influxdb_user} with password ${influxdb_password}."
+        iocage exec "${link_influxdb}" "curl -XPOST -u ${database_user}:${database_password} http://"${link_influxdb_ip4_addr%/*}":8086/query --data-urlencode 'q=CREATE DATABASE ${database}'"
+        echo "Database ${database} created with username ${database_user} with password ${database_password}."
```
Comment (Ornias1993) [inline blueprints/unifi/install.sh:54]: remove the influxdb -> database.

File: blueprints/unifi/install.sh [hunk 8]
```diff
@@ -73,19 +69,18 @@ else
   cp "${includes_dir}"/up.conf /mnt/"${global_dataset_config}"/"${1}"
   cp "${includes_dir}"/rc/unifi_poller.rc /mnt/"${global_dataset_iocage}"/jails/"${1}"/root/usr/local/etc/rc.d/unifi_poller
   chmod +x /mnt/"${global_dataset_iocage}"/jails/"${1}"/root/usr/local/etc/rc.d/unifi_poller
-  iocage exec "${1}" sed -i '' "s|influxdbuser|${influxdb_user}|" /config/up.conf
-  iocage exec "${1}" sed -i '' "s|influxdbpass|${influxdb_password}|" /config/up.conf
-  iocage exec "${1}" sed -i '' "s|unifidb|${influxdb_database}|" /config/up.conf
+  iocage exec "${1}" sed -i '' "s|influxdbuser|${database_user}|" /config/up.conf
+  iocage exec "${1}" sed -i '' "s|influxdbpass|${database_password}|" /config/up.conf
+  iocage exec "${1}" sed -i '' "s|unifidb|${database}|" /config/up.conf
```
Comment (Ornias1993) [inline blueprints/unifi/install.sh:74]: no

File: blueprints/unifi/install.sh [hunk 9]
```diff
@@ -73,19 +69,18 @@ else
...
   iocage exec "${1}" sed -i '' "s|unifiuser|${poller_user}|" /config/up.conf
   iocage exec "${1}" sed -i '' "s|unifipassword|${poller_password}|" /config/up.conf
-  iocage exec "${1}" sed -i '' "s|dbip|http://${DB_IP}:8086|" /config/up.conf
+  iocage exec "${1}" sed -i '' "s|dbip|http://${link_influxdb_ip4_addr%/*}:8086|" /config/up.conf
```
Comment (Ornias1993) [inline blueprints/unifi/install.sh:77]: nice work here.

File: blueprints/unifi/install.sh [hunk 10]
```diff
@@ -73,19 +69,18 @@ else
...
 
 
   iocage exec "${1}" sysrc unifi_poller_enable=YES
   iocage exec "${1}" service unifi_poller start
 
   echo "Please login to the Unifi Controller and add ${poller_user} as a read-only user."
-  echo "In Grafana, add Unifi-Poller as a data source."
```
Comment (Ornias1993) [inline blueprints/unifi/install.sh:88]: I suggest:

To automatically add this to grafana, add `link_influxdb` and `link_unifi` to the grafana jail configuration in config.yml

File: blueprints/unifi/update.sh [hunk 11]
```diff
@@ -5,7 +5,7 @@
 #init jail
 initblueprint "$1"
 
-if [[ ! "${poller}" ]]; then
+if [[ "${poller}" == "false" ]]; then
```
Comment (Ornias1993) [inline blueprints/unifi/update.sh:8]: correct syntax is:
```suggestion
if [[ -z "${poller}" ]]; then
```

**Comment by Ornias1993** [comment]
@colemamd #175 should be fixed now.

**Comment by colemamd** [comment]
I found the database discussion, but can't find where it's all documented. When I first started, I was under the impression the goal was to have the ability to have a fully automated way to connect services to database, and grafana. Did this change?

File: blueprints/unifi/install.sh (hunk 7 above)
Comment (colemamd) [reply blueprints/unifi/install.sh:54]: This section creates the database in influxdb. There isn't a db to connect to until this runs

**Comment by Ornias1993** [comment]
> I found the database discussion, but can't find where it's all documented. When I first started, I was under the impression the goal was to have the ability to have a fully automated way to connect services to database, and grafana. Did this change?

No, we just discussed it wan't okey to name it "database". It should be named influxdb, don't mess with it.

File: blueprints/unifi/install.sh (hunk 7 above)
Comment (Ornias1993) [reply blueprints/unifi/install.sh:54]: Im asking to stop renaming influxdb to database...

**Comment by Ornias1993** [comment]
This also needs to go into minor-dev branch btw.

**Comment by colemamd** [comment]
Closed. New PR #179

**Pull request closed without merging**